x-pack/platform/plugins/shared/embeddable_alerts_table/public/embeddable_alerts_table_factory.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { CoreStart } from '@kbn/core-lifecycle-browser';
+import type { ReactEmbeddableFactory } from '@kbn/embeddable-plugin/public';
+import {
+  initializeTimeRange,
+  initializeTitleManager,
+  useFetchContext,
+} from '@kbn/presentation-publishing';
+import { AlertsTable } from '@kbn/response-ops-alerts-table';
+import { AlertActionsCell } from '@kbn/response-ops-alerts-table/components/alert_actions_cell';
+import { getTime } from '@kbn/data-plugin/common';
+import { ALERT_TIME_RANGE, TIMESTAMP } from '@kbn/rule-data-utils';
+import { BehaviorSubject } from 'rxjs';
+import type { StartDependencies } from './types';
+import { EMBEDDABLE_ALERTS_TABLE_ID, LOCAL_STORAGE_KEY_PREFIX } from './constants';
+import type {
+  EmbeddableAlertsTableApi,
+  EmbeddableAlertsTableRuntimeState,
+  EmbeddableAlertsTableSerializedState,
+} from './types';
+
+export const getEmbeddableAlertsTableFactory = (
+  core: CoreStart,
+  deps: StartDependencies
+): ReactEmbeddableFactory<
+  EmbeddableAlertsTableSerializedState,
+  EmbeddableAlertsTableRuntimeState,
+  EmbeddableAlertsTableApi
+> => ({
+  type: EMBEDDABLE_ALERTS_TABLE_ID,
+  deserializeState: (state) => {
+    return state.rawState as EmbeddableAlertsTableSerializedState;
+  },
+  buildEmbeddable: async (state, buildApi, uuid) => {
+    const timeRange = initializeTimeRange(state);
+    const titleManager = initializeTitleManager(state);
+    const queryLoading$ = new BehaviorSubject<boolean | undefined>(true);
+    const { data, fieldFormats, licensing } = deps;
+    const { http, application, notifications, settings } = core;
+    const api = buildApi(
+      {
+        ...timeRange.api,
+        ...titleManager.api,
+        dataLoading$: queryLoading$,
+        serializeState: () => {
+          return {
+            rawState: { ...titleManager.serialize(), ...timeRange.serialize() },
+          };
+        },
+      },
+      { ...titleManager.comparators, ...timeRange.comparators }
+    );
+
+    return {
+      api,
+      Component: () => {
+        const { timeRange: selectedTimeRange } = useFetchContext(api);
+
+        return (
+          <AlertsTable
```

Comment:
Should we wrap the component with context providers like the `KibanaRenderContextProvider` and the `KibanaContextProvider`?

Reply:
We might not need this actually, let's maybe see how complex the wrapping logic around the table gets? (Doesn't seem too necessary from the draft I'm working on 🙂)